artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/impl/PagingStoreImpl.java

```diff
@@ -713,6 +713,9 @@ public boolean checkMemory(boolean runOnFailure, final Runnable runWhenAvailable
          }
       } else if (pagingManager.isDiskFull() || addressFullMessagePolicy == AddressFullMessagePolicy.BLOCK && (maxSize != -1 || usingGlobalMaxSize)) {
          if (pagingManager.isDiskFull() || maxSize > 0 && sizeInBytes.get() >= maxSize || pagingManager.isGlobalFull()) {
+            if (runWhenBlocking != null) {
+               runWhenBlocking.run();
```

Comment:
I don't have time to handle this one now.. I have to finish this PR today.

Reply:
Its unclear to me that you can trust this PR without changing it, given the fact it would be running tasks when it shouldnt, and these are now coordinating on important state.

Reply:
I will address this on a next PR if there's an issue here.